crates/loader/src/bindle/assets.rs

```diff
@@ -108,3 +120,24 @@ impl Copier {
         Ok(())
     }
 }
+
+async fn check_existing_file(path: &PathBuf, label: &Label) -> Result<bool> {
+    if let Ok(mut existing) = fs::File::open(path).await {
+        let mut buf = [0; 0xffff];
+        let mut hasher = Sha256::new();
+        let hex_sum = loop {
```

Comment:
The publish crate already has a function for the hash string (`file_digest_string`).  We probably don't want this crate to depend on `publish` but I'd suggest we copy it rather than have a separate inline implementation.  (I'm happy if you want to replace the implementation with this one, as long as we keep the same implementation in both copies.)